Assets/MixedRealityToolkit.Extensions/SceneTransitionService/SceneTransitionService.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace Microsoft.MixedReality.Toolkit.Extensions.SceneTransitions
 {
-    [MixedRealityExtensionService(SupportedPlatforms.WindowsStandalone|SupportedPlatforms.MacStandalone|SupportedPlatforms.LinuxStandalone|SupportedPlatforms.WindowsUniversal)]
+    [MixedRealityExtensionService(SupportedPlatforms.WindowsStandalone|SupportedPlatforms.MacStandalone|SupportedPlatforms.LinuxStandalone|SupportedPlatforms.WindowsUniversal, requiredProfileName: "MixedRealityRegisteredServiceProvidersProfile")]
```

Comment:
Using a single string isn't very flexible.  It's also a 1:1 match between a dataprovider/service and a profile without any extension. There is also a sort of weird redundancy here. Profiles determine what services they are valid for via the MixedRealityServiceProfile attribute. 

So...dataproviders are pointing at a specific profile and profiles are pointing at a specific service type. Also weirdly one direction is generally using concrete types while the other direction is interfaces. 

The system in general could be greatly cleaned up. But to reduce complexity, redundancy, and thus potential future errors, let's just stick to a boolean requiresProfile